disk/disk_linux.go

```diff
@@ -285,81 +285,99 @@ func PartitionsWithContext(ctx context.Context, all bool) ([]PartitionStat, erro
 		return nil, err
 	}
 
-	ret := make([]PartitionStat, 0, len(lines))
+	var ret []PartitionStat
+	if useMounts { // use mounts file
+		ret = parseFieldsOnMounts(lines, all, fs)
+	} else { // use mountinfo
+		ret, err = parseFieldsOnMountinfo(ctx, lines, all, fs, filename)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing mountinfo file %s: %w", filename, err)
+		}
+	}
 
+	return ret, nil
```

Comment:
This can be simplified, it would be more readable


```suggestion
	if useMounts { // use mounts file
		ret = parseFieldsOnMounts(lines, all, fs)
		return ret, nil
	}

	// use mountinfo
	ret, err = parseFieldsOnMountinfo(ctx, lines, all, fs, filename)
	if err != nil {
		return nil, fmt.Errorf("error parsing mountinfo file %s: %w", filename, err)
	}
 
	return ret, nil
```